infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/MachineResolver.java

```diff
@@ -116,18 +116,42 @@ private void normalizeMemory(Container container, InternalMachineConfig machineC
     if (ramRequest == 0) {
       machineConfig.getAttributes().put(MEMORY_REQUEST_ATTRIBUTE, defaultSidecarMemoryRequestBytes);
     }
+    String overriddenSidecarMemRequest = component.getMemoryRequest();
+    if (!isNullOrEmpty(overriddenSidecarMemRequest)) {
+      machineConfig
+          .getAttributes()
+          .put(
+              MEMORY_REQUEST_ATTRIBUTE,
+              Long.toString(KubernetesSize.toBytes(overriddenSidecarMemRequest)));
+    }
   }
 
   private void normalizeCpu(Container container, InternalMachineConfig machineConfig) {
     float cpuLimit = Containers.getCpuLimit(container);
     if (cpuLimit == 0) {
       machineConfig.getAttributes().put(CPU_LIMIT_ATTRIBUTE, defaultSidecarCpuLimitCores);
     }
+    String overriddenSidecarCpuLimit = component.getCpuLimit();
+    if (!isNullOrEmpty(overriddenSidecarCpuLimit)) {
+      machineConfig
+          .getAttributes()
+          .put(
+              CPU_LIMIT_ATTRIBUTE,
+              Float.toString(KubernetesSize.toCores(overriddenSidecarCpuLimit)));
```

Comment:
Since `KubernetesSize.toCores` and `KubernetesSize.toBytes` can throw an `IllegalArgumentException`, an invalid cpu or memory limit will cause the workspace to fail with 
```
Error: Failed to start the workspace: "Internal Server Error occurred, error time:
```
It isn't strictly an issue introduced in this PR, so it could be resolved later, but this is very bad UX. We should add validation methods (or checked exceptions) to KubernetesSize and use those to report to the user what went wrong.

Reply:
Surprisingly, but looks loke we're processing it in a right way on dashboard:
![image](https://user-images.githubusercontent.com/1651062/75763590-21d5ae80-5d45-11ea-9093-9b02063365ba.png)

Reply:
Ah that's funny -- I was testing this using a ~1 week old deployment of nightly (but still 7.10-SNAPSHOT) and could saw the internal server error, but on the latest nightly it correctly reports the issue. I must have missed a change somewhere :)